src/axom/primal/operators/detail/clip_impl.hpp

```diff
@@ -671,6 +724,116 @@ AXOM_HOST_DEVICE Polyhedron<T, NDIMS> clipTetrahedron(
   return poly;
 }
 
+/*!
+ * \brief Clips a 2D subject polygon against a clipping plane in 2D, returning
+ *        their geometric intersection as a polygon.
+ *
+ * \param inputList The input polygon to be clipped.
+ * \param plane The plane being used for clipping.
+ * \param eps The tolerance used for intersection.
+ *
+ * \return A clipped polygon.
+ */
+AXOM_SUPPRESS_HD_WARN
+template <typename PolygonType, typename PlaneType>
+AXOM_HOST_DEVICE PolygonType clipPolygonPlaneSimple(const PolygonType& inputList,
+                                                    const PlaneType& plane,
+                                                    double eps)
+{
+  using T = typename PolygonType::PointType::CoordType;
+  using SegmentType = Segment<T, 2>;
+  using PointType = Point<T, 2>;
+
+  PolygonType outputList;
+
+  const int numVertices = inputList.numVertices();
+  for(int iVert = 0; iVert < numVertices; iVert++)
+  {
+    const int prevVert = ((iVert - 1) == -1) ? (numVertices - 1) : (iVert - 1);
+    const PointType& current_point = inputList[iVert];
+    const PointType& prev_point = inputList[prevVert];
+
+    T seg_param;
+    PointType intersecting_point;
+    SegmentType subject_edge(prev_point, current_point);
+
+    const bool intersected = intersect(plane, subject_edge, seg_param, eps);
+    if(intersected)
+    {
+      intersecting_point = subject_edge.at(seg_param);
+    }
+
+    const int cur_p_orientation = plane.getOrientation(current_point, eps);
+    const int prev_p_orientation = plane.getOrientation(prev_point, eps);
+
+    if(cur_p_orientation == ON_POSITIVE_SIDE)
+    {
+      if(prev_p_orientation != ON_POSITIVE_SIDE)
+      {
+        outputList.addVertex(intersecting_point);
+      }
+      outputList.addVertex(current_point);
+    }
+    else if(prev_p_orientation == ON_POSITIVE_SIDE)
+    {
+      // In this branch, there has been a plane crossing from positive to negative.
+      if(intersected)
+      {
+        outputList.addVertex(intersecting_point);
+      }
+      else
+      {
+        // If there was not an intersection point then it was close but not
+        // within eps tolerance. It would have been really close to current_point.
+        // Since there was a plane crossing, add current_point.
+        outputList.addVertex(current_point);
+      }
+    }
+  }
+
+  return outputList;
+}
+
+/*!
+ * \brief Remove duplicate points in a polygon.
```

Comment:
Slight clarification:
```suggestion
 * \brief Remove duplicate adjacent points in a polygon.
```

Reply:
Fixed.